### PR TITLE
Expansion of Plotly options and add support for configuration options 

### PR DESCRIPTION
**Pull request by LMW38**
@essenciary, @hhaensel:

With this PR, I propose additions and changes for StipplePlotly. Note that some changes break backward compatibilty.

Here are the most significant changes and additions:

- In general, I propose to follow the original design pattern as laid out for PlotData: a mutable structure, and show, Dict and Stipple.render functions with optionals! as a supporting function. I followed the Plotly reference as much as possible, so names of options differ from Julia Plots.
- Plotly JS comes with a comprehensive set of defaults for graphs. So, it is not necessary to change and send from back-end to front-end all optional parameters/settings. This was how it was implemented for PlotData. I propose to apply it for PlotLayout as well. 
-  Plotly JS has several (sub)elements, e.g. axis, colorbar, legend, line marker, title etc. In this PR I promoted such elements to structures in Julia. Previous PR implemented this for plot annotations as demonstrated in example heatmap2.jl in the docs directory. I updated all examples to reflect the changes. 
- It is worthwhile for some use cases to change Plotly configuration options, e.g. change download filetype. This is the Config structure in Julia Charts.jl. However, this needs a change to vueplot.js by adding a third argument to javascript function newPlot. By any measure I am not familiar with javascript, so help is welcome here.
- In the docs are now examples for subplots and bar-type plots. I updated all other examples.

On the downside, creating code in Julia can become verbose for simple plots. I think one can create shortcuts or implement "styles" like PlotlyBase does.

Finally, Plotly JS has so many options that I still haven't implemented. So far, I've implemented plots that serve my use cases. If some graphs are missing, I am happy to create examples for the docs. Interaction with Vue (e.g. scaling of graph size) I have not explored yet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 /Manifest.toml
 .DS_Store
 empty1.jl
-empty2.jl
+customaxis.jl
```
Comment (essenciary) [inline .gitignore:4]: What are these entries for?

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "StipplePlotly"
 uuid = "ec984513-233d-481d-95b0-a3b58b97af2b"
 authors = ["Adrian <e@essenciary.com> and contributors"]
-version = "0.1.0"
+version = "0.1.1"
```
Comment (essenciary) [inline Project.toml:4]: I think we should bump to v2, given that it's breaking

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "StipplePlotly"
 uuid = "ec984513-233d-481d-95b0-a3b58b97af2b"
 authors = ["Adrian <e@essenciary.com> and contributors"]
```
Comment (essenciary) [inline Project.toml:3]: Please add yourself as an author here if you want :-)

**Comment by essenciary** [comment]
@LMW38 Thank you so much Leo, looking great! 

Tbh, you're the top authority for StipplePlotly :) If you think the code is ready to be merged, please do. 

Changes makes sense - yes, it does become a bit more verbose in places with the introduction of the new types but I can't see how we can avoid that, the Plotly.js API itself is quite complex. 

Happy to modify the JS code if you can give me more details please? Happy to have a chat or a quick call to discuss this. 

Unrelated: one thing that I'm curious about (something I've thinking about quite a bit) is how to handle the nested properties. Defining so many fields for custom types takes a lot of work. It's kind of nice cause it's self documenting in a way - but using Dicts (like PlotlyJS.jl does) has the benefit of only defining the keys that are needed, and for everything else defaults in JS are used.

File: .gitignore (hunk 1 above)
Comment (LMW38) [reply .gitignore:4]: These are templates from which I derive the examples.

File: Project.toml (hunk 2 above)
Comment (LMW38) [reply Project.toml:4]: Ok, I will do so.

File: Project.toml (hunk 3 above)
Comment (LMW38) [reply Project.toml:3]: OK, I will add my name.

**Comment by LMW38** [comment]
@essenciary 
Thank you for reviewing and suggestions. Before merging, we should discuss the modifications to JS code. I will contact you.

**Pull request merged**